projects/rsm/rsm_experiment.py

```diff
@@ -590,22 +594,42 @@ def _do_prediction(
                 # Uniform smoothing enabled
                 mass_pct = self.unif_smoothing
                 predictor_mass_pct -= mass_pct
-                predictions += mass_pct * torch.ones_like(predictor_outputs) / self.vocab_size
-
-            predictions += predictor_mass_pct * predictor_outputs
-
-            if not self.predictor_log_softmax:
-                prediction_log_probs = predictions.log()
-            else:
-                prediction_log_probs = predictions
-
-            pred_loss = self.predictor_loss(prediction_log_probs, pred_targets)  # NLLLoss
-            _, class_predictions = torch.max(predictor_outputs, 1)
+                predictions += mass_pct * torch.ones_like(predictor_dist) / self.vocab_size
+
+            predictions += predictor_mass_pct * predictor_dist
+
+            # No longer needed:
+            # if not self.predictor_log_softmax:
+            #     prediction_log_probs = predictions.log()
+            # else:
+            #     prediction_log_probs = predictions
+
+            # This loss is without inference-time model interpolation
+            pred_loss = self.predictor_loss(predictor_logits, pred_targets)  # cross-entropy loss
+
+            # This loss is for the interpolated model
+            predictor_dist_size = list(predictor_dist.size())
+            #print('shape dist: ', list(predictor_dist_size))
+            #print('shape tgts: ', list(pred_targets.size()))
+            # Alternate: torch.nn.functional.one_hot(x)
+            # labels_one_hot = torch.FloatTensor(predictor_dist_size)
+            # labels_one_hot = labels_one_hot.to(self.device)
+            # labels_one_hot.zero_()
+            # labels_one_hot.scatter_(1, pred_targets.unsqueeze(1), 1)
+            num_classes = predictor_dist_size[1]
+            labels_one_hot = torch.nn.functional.one_hot(pred_targets, num_classes=num_classes)
+            labels_one_hot = labels_one_hot.to(self.device).float()
+            #ll = (labels_one_hot * torch.log(predictor_dist)).sum(dim=[1]).mean()  This matches the training loss when other models are disabled
+            ll = (labels_one_hot * torch.log(predictor_dist)).sum(dim=[0,1])
```

Comment:
`predictor_dist` doesn't include the interpolated uniform/cache distributions since those are added into `predictions`. Should this line not be using `predictions` to give us a separate interpolated model loss?

Reply:
You're right, it should be predictions. Do you want me to make a new PR?